src/core/qgsaction.h

```diff
@@ -191,6 +191,20 @@ class CORE_EXPORT QgsAction
      */
     void writeXml( QDomNode &actionsNode ) const;
 
+    /**
+     * Sets an expression context scope to use for running the action.
+     *
+     * \since QGIS 3.0
+     */
+    void setExpressionContextScope( const QgsExpressionContextScope &scope );
```

Comment:
I don't have the implementation in mind completely right now, please bear with me ;)

Is there a QgsAction for each menuitem (or other means to trigger it) or a (shared) one in the registry? I.e. is there a risk of concurrency / race condition when an action is displayed in the feature form and the attribute table at the same time and both use this setter for different scopes on the same action?

Reply:
According to the `QgsActionMenu::reloadActions` method, it seems that there's well a QgsAction for each item in the menu. So I don't think there's a risk of concurrency in this case.

Reply:
Thanks, good for me then